src/analysis/scoring.cc

```diff
@@ -0,0 +1,305 @@
+#include "analysis/scoring.h"
+#include <algorithm>
+#include <vector>
+
+#include "analysis/analysis.h"
+#include "analysis/util.h"
+#include "analysis/yakus.h"
+#include "types/gamestate.h"
+#include "types/pieces.h"
+#include "types/piecetype.h"
+#include "types/score.h"
+
+namespace mahjong {
+
+namespace {
+const int kLowHanBasicPoints = 2000;
+const int kManganBasicPoints = 2000;
+const int kManganMinHan = 5;
+const int kHanemanBasicPoints = 3000;
+const int kHanemanMinHan = 6;
+const int kBaimanBasicPoints = 4000;
+const int kBaimanMinHan = 8;
+const int kSanbaimanBasicPoints = 6000;
+const int kSanbaimanMinHan = 11;
+const int kYakumanBasicPoints = 8000;
+
+const int kPinfuDiscard = 30;
+const int kPinfuSelfdraw = 20;
+const int kConcealedDiscard = 25;
+const int kOpenOrSelfdraw = 20;
+
+const int kSimplepon = 2;
+const int kTermHonorpon = 4;
+const int kSimplekan = 8;
+const int kTermHonorkan = 16;
+
+const int kConcealedMul = 2;
+const int kCsimplepon = kSimplepon * kConcealedMul;
+const int kCtermHonorpon = kTermHonorpon * kConcealedMul;
+const int kCsimplekan = kSimplekan * kConcealedMul;
+const int kCtermHonorkan = kTermHonorkan * kConcealedMul;
+
+const int kDragonseatprevalentwind = 2;
+const int kEdgeclosedpairwait = 2;
+const int kSelfdraw = 2;
+const int kOpenpinfu = 2;
+
+const int kLowedgewait = 3;
```

Comment:
is there some better data structure that makes sense for this? I suppose being able to do constant math ops is pretty nice for things like concealed

Reply:
Yea I want to break a lot of the data out of the code and into like toml files or something.